### PR TITLE
react-react1-week3

### PR DESCRIPTION
**Pull request by Merhawi-gebreslasie**

### CODE REVIEW

File: react/react1/week3/homework/shift-app/src/usercomponents/UserList.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+const UserList = ({ fetchUrl }) => {
+  const [users, setUsers] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    setLoading(true);
+    const timeOut = setTimeout(getResult, 3000);
+    return () => clearTimeout(timeOut);
+  }, []);
```
Comment (semih1239) [inline react/react1/week3/homework/shift-app/src/usercomponents/UserList.js]: You dont need to set a timeout here. You can call getResult function directly here. Also getResult ils already including Loading statement.
```suggestion
  useEffect(() => {
    getResult()
  }, []);

```

File: react/react1/week3/homework/shift-app/src/usercomponents/UserList.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    });
+  };
+
+  const handleFilter = (search) => {
+    setUsers((prevValues) =>
+      prevValues.filter((prev) =>
+        prev.name.toLowerCase().includes(search.toLowerCase())
+      )
+    );
+  };
```
Comment (semih1239) [inline react/react1/week3/homework/shift-app/src/usercomponents/UserList.js]: You can add getResult Function in handleFilter. Your solution is deleting other users from statement and we can search for only one time. With this solution you will set all users before search.
```suggestion
  const handleFilter = async (search) => {
    const users = await getResult()
    setUsers((prevValues) =>
      prevValues.filter((prev) =>
        prev.name.toLowerCase().includes(search.toLowerCase())
      )
    );
  };

```

File: react/react1/week3/homework/todo-app/src/components/AddTodo.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import React, { useState } from "react";
+
+const AddTod = ({ addTodo, items }) => {
+  const newId = Math.floor(Math.random() * 10000);
```
Comment (semih1239) [inline react/react1/week3/homework/todo-app/src/components/AddTodo.js]: In the projects it is not a good idea to use random number for ids. You can get last item id from todolist, you can add 1 to it and you can set it when you call this component.
```suggestion
const AddTod = ({ addTodo, items, newId }) => {
```

File: react/react1/week3/homework/todo-app/src/components/TodoList.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+export const TodoList = ({ fecthUrl }) => {
+  const [items, setItems] = useState([]);
+  const [error, setError] = useState(null);
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    setLoading(true);
+    const timeOut = setTimeout(getData, 2000);
+    return () => clearTimeout(timeOut);
```
Comment (semih1239) [inline react/react1/week3/homework/todo-app/src/components/TodoList.js]: Same here. You dont need to set timeout here you can call directly function here
```suggestion
    getData()
```

**Review by semih1239 — COMMENTED** [review]
You did very good job with Shift and Todo applications. Good Job 💪.
I wrote some small comments to below

File: react/react1/week3/homework/shift-app/src/usercomponents/UserList.js (hunk 1 above)
Comment (Merhawi-gebreslasie) [reply react/react1/week3/homework/shift-app/src/usercomponents/UserList.js]: Thanks ,  I have just implemented the changes,  I was trying to show the loading text  a little longer with timeout.

File: react/react1/week3/homework/shift-app/src/usercomponents/UserList.js (hunk 2 above)
Comment (Merhawi-gebreslasie) [reply react/react1/week3/homework/shift-app/src/usercomponents/UserList.js]: thank you so much , I  did not notice that and that was very interesting .

File: react/react1/week3/homework/todo-app/src/components/AddTodo.js (hunk 3 above)
Comment (Merhawi-gebreslasie) [reply react/react1/week3/homework/todo-app/src/components/AddTodo.js]: yes thanks. changes implemented.

File: react/react1/week3/homework/todo-app/src/components/TodoList.js (hunk 4 above)
Comment (Merhawi-gebreslasie) [reply react/react1/week3/homework/todo-app/src/components/TodoList.js]: Thanks, changes implemented.

**Pull request merged**